api/review_checklist.md

```diff
@@ -0,0 +1,137 @@
+# API Review Checklist
+
+This checklist is intended to be used when reviewing xDS API changes.
+Users who wish to contribute API changes should read this and proactively
+consider the answers to these questions before sending a PR.
+
+## Feature Enablement
+- Are the default values going to cause behavior changes for existing users
+  who do not know about the change and have not updated the resources being
+  served by their control plane?
+  - If yes, do we have some estimate of how many users will be affected?
+  - Why is it justified to change the default behavior, rather than making
+    this feature opt-in?
+    - Some possible justifications include security concerns with existing
+      behavior, or a desire to eliminate legacy behavior.
+  - What is the plan to make this change in a safe way?  For example, is the
+    transition going to be staged over the course of several minor xDS versions?
+  - How will we warn users about this change?
+    - Possible ways to do this include release notes, announcements, warnings
+      from the code, etc.
+- Will users have a way to disable this change if it causes problems?
+  - If not, why do we think that's okay?  (It might be the case that we think
+    it will not actually affect anyone, or no one will care.)
+  - If so, is the mechanism to disable it part of the xDS API, or is it
+    acceptable to have a separate knob for this in the client?  (See also
+    "Genericness" below -- if this is not part of the API, will every xDS
+    client need to add a different knob?  Is consistency across clients
+    important for this?)
+
+## Style
+- If an enum is being introduced, should this be a oneof with empty messages
+  for future API growth?
+- When a new field is introduced for a distinct purpose, should this be a
+  message to allow for future API growth?
+- For time related fields, is a Duration or Timestamp WKT used rather than
+  raw integers for seconds etc?
+- Is the PR aligned with the [API style guidelines](STYLE.md)?
+
+## Validation Rules
+- Is the field mandatory? Does it have the required rule?
```

Comment:
I wonder if we can enforce any of these with fix_format / proto scripts.  If we had a required required/not_required annotation, or required "unbounded" annotations, so folks had to explicitly annotate, it'd be annoying but might catch more.
I'd file this under "food for thought" where you and the other shephards can decide if it's worth it after seeing what slips through review over time :-)

Reply:
I'll let @htuch think about this one.

Reply:
This could be done. I think if I had my wishtlist of tooling, we would have something that during code review annotates the PR when a new field is introduced and is missing constraints to verify. 

@markdroth can you be more explicit and say something like "Is the field mandatory? If so, are protocgen-validate constraints defined to enforce this?".

Reply:
I added a separate bullet about having PGV rules, since that basically applies to all of the bullets in this section.